site/learn/expert-advice/authentication/webauthn-explained.md

```diff
@@ -0,0 +1,478 @@
+
+As mentioned above, each passkey is tied to a specific domain name and only allowed on HTTPS URLs, which are enforced by the browser. For instance, you can't do an authentication request for the site `example.com` if the page is served by `cosmosclownstore.com`. If these security checks pass, the browser passes the request to the authenticator via CTAP2. The authenticator then authenticates the user using biometric means such as facial recognition (or any other means it implements).
+
+The user can choose not to authenticate with the authenticator, in which case it sends back an error message. If the user does authenticate, the authenticator creates a signed assertion and passes it back to the browser. The browser passes the assertion back to Cosmo's Clown Store.
+
+Cosmos Clown Store can verify the assertion with the public key that is stored and associated with that user. If the assertion checks out, the user is logged in and can buy all the clownwear they want.
+
+## Some WebAuthn Code
+
+The core of the web authentication workflow is using the credentials API. To begin authentication, use the `get` method.
+
+```javascript
+navigator.credentials.get(options)
+  .then(function (assertionResponse) {
+    // Send authentication status to server
+  }).catch(function (err) {
+    // No acceptable authenticator or user refused.
+  });
+```
+
+You pass it an `options` object and get a response back. Here's what the options object looks like:
+
+```javascript
+options = {
+  publicKey: {
+    rpId: "cosmosclownstore.com",
+    challenge: "....",
+    userVerification: "preferred",
+    //...
+  }
+}
+```
+
+The options object has a few important properties:
+
+- the `publicKey` top-level key: this indicates we're using WebAuthn
+- the `rpId`: this tells the authenticator which website this request is for, such as cosmosclownstore.com
+- the `challenge`: a random string to prevent replay attacks
+
+There are other options as well, which allow the site to determine if the user should be fully verified and authenticated or just prove presence. That option will be discussed more below.
+
+After you call `get` with an `options` object, the response is returned as a promise. Here's an example response.
+
+```javascript
+{
+  authenticatorData: {
+    rpIdHash : ..., // hash of the requesting party Id
+    flags : ..., // bit array including authentication results
+    // other info
+  },
+  clientDataJSON: {
+    type: "webauthn.get",
+    challenge: ... // should match initial challenge
+  }
+}
+```
+
+If the response is successful, the response data will have an `authenticatorData` key, which has a `flags` member. `flags` is a bit array indicating whether the user authenticated or not, as well as what assurance you have. 
```

Comment:
I'm not sure what you mean by "indicating whether the user authenticated or not." The response is only returned from the authenticator if the user has completed the ceremony. I'd say the most important thing on the [flags](https://www.w3.org/TR/webauthn-2/#flags) is whether user verification was used during the ceremony. Is that what you mean by "authenticated?"